content/Refactoring - Sebastian Herrera.md

```diff
@@ -0,0 +1,144 @@
+
+I will present just 2 simple refactoring as examples. I don’t think the purpose of this article should be for you to learn many of them or learning some complex ones. I believe that is something you can learn by yourself in somewhere else, but you will do it only if you feel enough motivation to do it.
+
+**First Example - Decompose Conditional Expression:**
+You have a complicated conditional (if-then-else) statement.
+→ Extract methods from the condition, “then” part, and “else” part.
+
+![Decompose Conditional Expression Example](images/decompose_conditional_expression.png)
+
+- Motivation: Conditional logic adds complexity to the code. You can make your intention clearer by decomposing it and replacing chunks of code with a method call named after the intention of that block of code.
+- Mechanic:
+  1.  Extract the conditions into their own methods.
+  2.  Test.
+  3.  Extract the actions inside the conditional code into their own methods.
+  4.  Test.
+
+**Second Example - Extract Class:**
+You have one class doing work that should be done by two.
+→ Create a new class and move the relevant fields and methods from the old class into the new class.
+
+![Exctact Class Example](images/extract_class_1.png)
+
+![Exctact Class Example](images/extract_class_2.png)
+
+![Exctact Class Example](images/extract_class_3.png)
+
+- Motivation: In practice, classes grow over time. One class with lots of methods and data is too big to understand easily, so it becomes hard to maintain.
+- Mechanic:
+  1. Decide how to split the responsibilities of the class.
+  2. Create a new class to express the split-off responsibilities. If the responsibilities of the old class no longer match its name, rename the old class.
+  3. Make a link from the old to the new class.
+  4. Use Move Field on each field you want to move.
+  5. Test after each move.
+  6. Use Move Method to move methods over from the old to the new. Start with lower-level methods and build to the higher level.
+  7. Test after each move.
+  8. Review and reduce the interfaces of each class.
+  9. Decide whether multiple clients will be able to access the class. If you do allow access to multiple clients, decide whether to expose the new class as a reference object or as an immutable value object.
+
+## 7) FINAL THOUGHTS
+In my own experience as a software engineer, I have seen many projects grow a lot over time, and by the hand, their code became harder to understand, harder to maintain, harder to work with and more prone to introducing bugs. And I know for as long as I work in this industry, I will continue to see this happen.
+
+There is no magic trick for that. But, so far I have also seen that refactoring the code as part of your daily programming, is probably the best technique you and your team can apply to counter those undesired effects.
+
+Give it a chance, and practice it until it just becomes a habit. I can assure that you will live the benefits of it.
```

Comment:
```suggestion
Give it a chance, and practice it until it just becomes a habit. I can assure you that you will live the benefits of it.
```

Reply:
I'm not completely sure about this one .. maybe will ask an english speaker 😅 . 
Not because of the meaning, but because of using the word `you` twice.